### PR TITLE
[FIX] base, tools: fixed an error caused sometimes when importing a p…

### PR DESCRIPTION
**Pull request by bfr-o**
…df without embedded files

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/tools/pdf.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ class OdooPdfFileReader(PdfFileReader):
     # OVERRIDE of PdfFileReader to add the management of multiple embedded files.
 
     def getAttachments(self):
-        if "/Names" not in self.trailer["/Root"]:
+        if "/Names" not in self.trailer["/Root"] or "/EmbeddedFiles" not in self.trailer["/Root"]["/Names"] or "/Names" not in self.trailer["/Root"]["/Names"]["/EmbeddedFiles"]:
```
Comment (odony) [inline odoo/tools/pdf.py:62]: `PyPDF2.generic.DictionaryObject` are dicts so we can use `d.get(key, default)`.
How about this more concise version? 

```suggestion
        if not self.trailer["/Root"].get("/Names", {}).get("/EmbeddedFiles", {}).get("/Names"):
```

Comment (bfr-o) [reply odoo/tools/pdf.py]: Much better, thanks !

**Comment by odony** [comment]
To follow our commit message conventions, I'd suggest something like this:

```
[FIX] tools.pdf: do not crash when PDF has no attachments

Depending on the PDF version and the presence of embedded files, the PDF
document catalog `trailer["/Root"]` may not have any Name Dictionary
`["/Names"]`, and the latter may not have any `EmbeddedFiles` entry.

In that case `getAttachments()` should return an empty list rather than
raising a `KeyError`.

Ref: Section 7.7.2, 7.7.4 and 7.11.4 of [PDF 1.7 spec](https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/PDF32000_2008.pdf)
```

**Comment by qdp-odoo** [comment]
@robodony r+

**Pull request closed without merging**